pkg/services/rpcsrv/server_test.go

```diff
@@ -2003,6 +2003,107 @@ func TestSubmitNotaryRequest(t *testing.T) {
+			c.ProtocolConfiguration.P2PSigExtensions = false
+		})
+		defer chain.Close()
+		defer rpcSrv.Shutdown()
+		t.Run("getrawnotarypool", func(t *testing.T) {
+			body := doRPCCallOverHTTP(fmt.Sprintf(rpcPool, "[]"), httpSrv.URL, t)
+			checkErrGetResult(t, body, true, neorpc.InternalServerErrorCode)
+		})
+		t.Run("getrawnotaryrequest", func(t *testing.T) {
+			body := doRPCCallOverHTTP(fmt.Sprintf(rpcRequest, " ", 1), httpSrv.URL, t)
+			checkErrGetResult(t, body, true, neorpc.InternalServerErrorCode)
+		})
+	})
+
+	chain, rpcSrv, httpSrv := initServerWithInMemoryChainAndServices(t, false, true, false)
+	defer chain.Close()
+	defer rpcSrv.Shutdown()
+
+	submitNotaryRequest := func(t *testing.T, fail bool, errCode int64, params ...string) func(t *testing.T) {
+		return func(t *testing.T) {
+			rpcSubmit := `{"jsonrpc": "2.0", "id": 1, "method": "submitnotaryrequest", "params": %s}`
+			ps := `[` + strings.Join(params, ",") + `]`
+			req := fmt.Sprintf(rpcSubmit, ps)
+			body := doRPCCallOverHTTP(req, httpSrv.URL, t)
+			checkErrGetResult(t, body, fail, errCode)
+		}
+	}
+
+	compareTx := func(t *testing.T, expectedTx, actualTx transaction.Transaction) func(t *testing.T) {
+		return func(t *testing.T) {
+			require.Equal(t, expectedTx.Hash(), actualTx.Hash())
+			require.Equal(t, expectedTx.Nonce, actualTx.Nonce)
+			require.Equal(t, expectedTx.Sender(), actualTx.Sender())
+			require.Equal(t, expectedTx.Signers, actualTx.Signers)
+			require.Equal(t, expectedTx.ValidUntilBlock, actualTx.ValidUntilBlock)
+			require.Equal(t, expectedTx.Script, actualTx.Script)
+			require.Equal(t, expectedTx.Attributes, actualTx.Attributes)
+		}
+	}
+
+	t.Run("test getrawnotarypool", func(t *testing.T) {
+		t.Run("empty pool", func(t *testing.T) {
+			body := doRPCCallOverHTTP(fmt.Sprintf(rpcPool, "[true]"), httpSrv.URL, t)
+			res := checkErrGetResult(t, body, false, 0)
+			actual := new(result.RawNotaryPool)
+			require.NoError(t, json.Unmarshal(res, actual))
+			require.Equal(t, 0, len(actual.NotaryHashes))
+		})
+
+		t.Run("submit first valid notary request", func(t *testing.T) {
+			sender := testchain.PrivateKeyByID(0) // owner of the deposit in testchain
+			notaryRequest = createValidNotaryRequest(chain, sender, 1)
+			nrBytes, err := notaryRequest.Bytes()
+			require.NoError(t, err)
+			str := fmt.Sprintf(`"%s"`, base64.StdEncoding.EncodeToString(nrBytes))
+			submitNotaryRequest(t, false, 0, str)(t)
+		})
+
+		t.Run("get notary pool & check tx hashes", func(t *testing.T) {
+			body := doRPCCallOverHTTP(fmt.Sprintf(rpcPool, "[true]"), httpSrv.URL, t)
+			res := checkErrGetResult(t, body, false, 0)
+			actual := new(result.RawNotaryPool)
+			require.NoError(t, json.Unmarshal(res, actual))
+			require.Equal(t, 1, len(actual.NotaryHashes))
+			require.Equal(t, notaryRequest.MainTransaction.Hash(), actual.NotaryHashes[0].MainTxHash)
+			require.Equal(t, notaryRequest.FallbackTransaction.Hash(), actual.NotaryHashes[0].FallbackTxHashes[0])
+		})
```

Comment:
Anna, could you please tell me how to get case when 2 fallback transactions are corresponding with 1 main transaction? I want to add test with something similar to:
{
    "main_hash_1": ["fallback_hash_1", "fallback_hash_2",..],
    "main_hash_2": ["fallback_hash_1", "fallback_hash_2",..]
}